pkg/skaffold/build/cache/cache.go

```diff
@@ -95,12 +96,23 @@ func NewCache(ctx context.Context, builder build.Builder, opts *skafconfig.Skaff
 		useCache:      opts.CacheArtifacts,
 		client:        client,
 		builder:       builder,
-		needsPush:     needsPush,
+		pushImages:    pushImages(cfg),
```

Comment:
nit: `cfg.LocalBuild != nil && cfg.LocalBuild.Push != nil && *cfg.LocalBuild.Push`

then you could remove `pushImages()` altogether, and code coverage won't decrease :)

Reply:
cool thanks!